pandatool/src/deploy-stub/deploy-stub.c

```diff
@@ -640,8 +694,21 @@ int main(int argc, char *argv[]) {
   ModuleDef *moddef;
   const char *log_filename;
   void *blob = NULL;
+#if defined(MS_WINDOWS)
+  char blob_path[MAX_PATH];
+#else
+  char blob_path[PATH_MAX];
+#endif
+  get_blob_path(argv[0], blob_path);
   log_filename = NULL;
-
+  // FreezeTool will overwrite blobinfo on non-MacOS to set the blob_offset 
+  // to non -1. If it is -1, then we need to load it from FileSystem
+  int load_external = 0;
+  if (blobinfo.blob_offset == -1) {
```

Comment:
Can we combine this with the later blob_offset check to avoid needing to define a new variable to block the later branch. Does this have to happen earlier for some reason?